### DISCUSSION TITLE
OSM Tags and Penalty Rules for Narrow Bidirectional Roads ?

### DISCUSSION THREAD

**Discussion by julcnx** [valhalla/valhalla#4591] [Q&A] [answered]
I’m in the midst of exploring Valhalla to understand the supported OSM tags for identifying narrow roads that permit bidirectional traffic in the default car profile. Additionally, I’m interested in the penalty rules:

Are there penalties for roads with a width less than X?
Are there any minimum requirements for width?
Does the number of lanes play a role?
Are there any other supported tags?
If you have any pointers to the relevant code, I’d greatly appreciate it! 🙏🙏🙏

**Comment by nilsnolde** [comment, marked as answer]
> Are there penalties for roads with a width less than X?
Are there any minimum requirements for width?

only as exclusive costing option, best look at the docs to understand more: https://valhalla.github.io/valhalla/api/turn-by-turn/api-reference/

> Does the number of lanes play a role?

I don't think for costing at all

> Are there any other supported tags?

see here: https://taginfo.openstreetmap.org/projects/valhalla#tags

on a broader topic, we'd like to have support for offroad stuff too, e.g. preferring or at least not penalizing bad surfaces, possibly an option to favor curvy roads etc.
